pkg/antctl/raw/multicluster/resourceimport.go

```diff
@@ -0,0 +1,201 @@
+		return
+	}
+}
+
+func init() {
+	command = &cobra.Command{
+		Use: "resourceimport",
+		Aliases: []string{
+			"resourceimports",
+			"ri",
+		},
+		Short:   "Print Multi-Cluster ResourceImports",
+		Args:    cobra.MaximumNArgs(1),
+		Example: resourceImportExamples,
+		RunE:    runE,
+	}
+	o := &resourceImportOptions{}
+	options = o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of ResourceImports")
+	command.Flags().StringVarP(&o.outputFormat, "output", "o", "", "Output format. Supported formats: json|yaml")
+	command.Flags().BoolVarP(&o.allNamespaces, "all-namespaces", "A", false, "If present, list ResourceImports across all Namespaces")
+
+	GetCmd.AddCommand(command)
+}
+
+func runE(cmd *cobra.Command, args []string) error {
+	options.validateAndComplete()
+	argsNum := len(args)
+	if options.allNamespaces && argsNum > 0 {
+		return fmt.Errorf("a resource cannot be retrieved by name across all Namespaces")
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	kubeconfig.GroupVersion = &schema.GroupVersion{Group: "", Version: ""}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+
+	scheme := k8sruntime.NewScheme()
+	err = mcsscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = antreamcscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = k8sscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	singleResource := false
+	if argsNum > 0 {
+		singleResource = true
+	}
+	var res interface{}
+
+	if singleResource {
+		resourceImportName := args[0]
+		resourceImport := multiclusterv1alpha1.ResourceImport{}
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: options.namespace,
+			Name:      resourceImportName,
+		}, &resourceImport)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return fmt.Errorf("ResourceImport %s not found in Namespace %s", resourceImportName, options.namespace)
+			}
+
+			return err
+		}
+
+		gvks, unversioned, err := k8sClient.Scheme().ObjectKinds(&resourceImport)
+		if err != nil {
+			return err
+		}
+		if !unversioned && len(gvks) == 1 {
+			resourceImport.SetGroupVersionKind(gvks[0])
+		}
+		res = resourceImport
+	} else {
+		resourceImportList := &multiclusterv1alpha1.ResourceImportList{}
+		err = k8sClient.List(context.TODO(), resourceImportList, &client.ListOptions{Namespace: options.namespace})
+		if err != nil {
+			return err
+		}
+
+		if len(resourceImportList.Items) == 0 {
+			if options.namespace != "" {
+				fmt.Printf("No resource found in Namespace %s\n", options.namespace)
+			} else {
+				fmt.Println("No resource found in all Namespaces")
+			}
+			return nil
+		}
+		res = resourceImportList.Items
+	}
+
+	return output(res, singleResource, options.outputFormat, resourceimport)
```

Comment:
It is tricky to define a nil global variable and use it as type. Could we follow tableOutputForGetCommands() instead? And any reason we cannot reuse the TableOutput interface, and even tableOutputForGetCommands?

Reply:
hi jianjun. I moved out the method `tableOutputForGetCommands` out of commandDefinition and made it public then  re-used it in `antctl mc get`. To avoid circle imports, I imported the resourceimport.go in main.go to initialize the command.
In the meanwhile, the JSON output and the YAML also re-used the method in commandDefinition. Thanks for your ocmment.

Reply:
Thanks for making the change!